File: app/src/main/java/com/example/cliptokindle/HttpApp.java
package com.example.cliptokindle;

import android.util.Log;

import java.io.IOException;
import java.util.List;
import java.util.Map;

import fi.iki.elonen.NanoHTTPD;

public class HttpApp extends NanoHTTPD {
    private static final StringBuilder msg = new StringBuilder();

    public HttpApp() {
        super(8080);
    }

    @Override
    public void start() throws IOException {
        start(NanoHTTPD.SOCKET_READ_TIMEOUT, true);
    }

    @Override
    public void stop() {
        super.stop();
    }

    @Override
    public Response serve(IHTTPSession session) {
        return newFixedLengthResponse(PageGenerator.generate());
        //return newFixedLengthResponse(msg.toString() + "</body></html>\n");
    }
    
    public static void add(Text text) {
        msg.append(text.toHtml());
    }
}

File: app/src/main/java/com/situ2001/cliptokindle/util/HttpApp.kt
package com.situ2001.cliptokindle.util

import fi.iki.elonen.NanoHTTPD
import java.io.IOException

class HttpApp(port: Int) : NanoHTTPD(port) {
    @Throws(IOException::class)
    override fun start() {
        start(SOCKET_READ_TIMEOUT, true)
    }

    override fun serve(session: IHTTPSession): Response {
        return newFixedLengthResponse(PageGenerator.getPageGenerator()?.generate())
        //return newFixedLengthResponse(msg.toString() + "</body></html>\n");
    }
}